### PR TITLE
(feat/be 10) reset password

### PR DESCRIPTION
**Pull request by MuhammadGb**
# General Changes

-   Implement OTP verification feature
-  Implement reset password functionality

##.

## Linear Link
https://linear.app/team-chisel-hng9/issue/BE-10/endpoint-to-reset-password

## Checklist

-   [ ] The base branch is set to `dev`
-   [ ] The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
-   [ ] The Linear issue has been linked to the PR
-   [ ] End-to-end tests(if any) are passing without any errors-   [ ] New third-party packages, if any, do not introduce potential security threats
-   [ ] There are no CI changes, or they have been OK’d by the devops team
-   [ ] Errors are handled using the right error handles

### CODE REVIEW

File: server/controllers/resetController.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		throw new BadRequestError("passwords must be similar");
+	}
+
+	const user = await User.findOne({ email: email });
+
+	if (!user) {
+		throw new BadRequestError("email not found in database");
+	}
+
+	const comparePassword = await user.compare(hashedPassword, user.password);
```
Comment (Destiny-01) [inline server/controllers/resetController.js:19]: What's hashed password here?

Comment (MuhammadGb) [reply server/controllers/resetController.js]: Fixed it now @Destiny-01

File: server/controllers/authController.js [hunk 2]
```diff
@@ -194,12 +195,60 @@ const forgotPassword = async (req, res, next) => {
...
+		throw new BadRequestError("confirm with a similar password");
+	}
+
+	const user = await User.findOne({ email: email });
+
+	if (!user) {
+		throw new BadRequestError("email not found in database");
+	}
+
+	const comparePassword = await user.compare(hashedPassword, user.password);
```
Comment (Destiny-01) [inline server/controllers/authController.js:213]: Are you comparing hashes? I don't understand that line

File: server/controllers/authController.js [hunk 3]
```diff
@@ -194,12 +195,60 @@ const forgotPassword = async (req, res, next) => {
...
+const resetPassword = async (req, res) => {
+	const { password, email, confirmPassword } = req.body;
+	const salt = bcrypt.genSaltSync(10);
+	const hashedPassword = bcrypt.hashSync(password, salt);
 
+	if (password !== confirmPassword) {
+		throw new BadRequestError("confirm with a similar password");
+	}
+
+	const user = await User.findOne({ email: email });
```
Comment (Destiny-01) [inline server/controllers/authController.js:207]: No need to add the second `:email`

File: server/controllers/authController.js (hunk 2 above)
Comment (MuhammadGb) [reply server/controllers/authController.js:213]: Yes @Destiny-01  the hashed password is being compared with the one fetched from the dB (which is also hashed)

**Pull request merged**